src/json_stream_gen.c

```diff
@@ -0,0 +1,1478 @@
+    }
+    else
+    {
+        if (!json_stream_gen_printf(
+                p_gen,
+                "%s%s%.*s"
+                "%" PRIu64,
+                p_gen->is_first_item ? "" : ",",
+                p_gen->p_eol,
+                p_gen->cur_nesting_level * p_gen->cfg.indentation,
+                p_gen->p_indent_filling,
+                val))
+        {
+            return false;
+        }
+    }
+    p_gen->is_first_item = false;
+    return true;
+}
+
+bool
+json_stream_gen_add_bool(json_stream_gen_t* const p_gen, const char* const p_name, const bool val)
+{
+    if (NULL != p_name)
+    {
+        if (!json_stream_gen_printf(
+                p_gen,
+                "%s%s%.*s\"%s\":%s"
+                "%s",
+                p_gen->is_first_item ? "" : ",",
+                p_gen->p_eol,
+                p_gen->cur_nesting_level * p_gen->cfg.indentation,
+                p_gen->p_indent_filling,
+                p_name,
+                p_gen->p_delimiter,
+                val ? "true" : "false"))
+        {
+            return false;
+        }
+    }
+    else
+    {
+        if (!json_stream_gen_printf(
+                p_gen,
+                "%s%s%.*s"
+                "%s",
+                p_gen->is_first_item ? "" : ",",
+                p_gen->p_eol,
+                p_gen->cur_nesting_level * p_gen->cfg.indentation,
+                p_gen->p_indent_filling,
+                val ? "true" : "false"))
+        {
+            return false;
+        }
+    }
+    p_gen->is_first_item = false;
+    return true;
+}
+
+bool
+json_stream_gen_add_null(json_stream_gen_t* const p_gen, const char* const p_name)
+{
+    if (NULL != p_name)
+    {
+        if (!json_stream_gen_printf(
+                p_gen,
+                "%s%s%.*s\"%s\":%s"
+                "%s",
+                p_gen->is_first_item ? "" : ",",
+                p_gen->p_eol,
+                p_gen->cur_nesting_level * p_gen->cfg.indentation,
+                p_gen->p_indent_filling,
+                p_name,
+                p_gen->p_delimiter,
+                "null"))
+        {
+            return false;
+        }
+    }
+    else
+    {
+        if (!json_stream_gen_printf(
+                p_gen,
+                "%s%s%.*s"
+                "%s",
+                p_gen->is_first_item ? "" : ",",
+                p_gen->p_eol,
+                p_gen->cur_nesting_level * p_gen->cfg.indentation,
+                p_gen->p_indent_filling,
+                "null"))
+        {
+            return false;
+        }
+    }
+    p_gen->is_first_item = false;
+    return true;
+}
+
+static char
+json_stream_gen_get_decimal_point(void)
+{
+    struct lconv* const p_lc = localeconv();
+    if (NULL == p_lc)
+    {
+        return '.';
+    }
+    return *p_lc->decimal_point;
+}
+
+typedef struct json_stream_gen_float_str_buf_t
+{
+    char buffer[JSON_STREAM_GEN_STR_BUF_SIZE_FLOAT];
+} json_stream_gen_float_str_buf_t;
+
+static bool
+json_stream_gen_float_to_str(
```

Comment:
I would add the decimal separator as an argument to the function creating the string. 

As a use-case example elsewhere, in the Ruuvi Cloud we convert floats to strings by user preference. The precision and locale comes from user settings, not system locale. Passing the separator as an argument gives end-user more flexibility to decide if they want to use system locale, user setting, fixed "." or fixed ",".

Most importantly, this change means that the output will not depend on system running the code unless user explicitly wants it to depend on locale.

Reply:
I checked the standards related to JSON format and they state that only '.' can be used as a decimal separator (so, comma is not allowed).
![image](https://github.com/ruuvi/ruuvi.json_stream_gen.c/assets/4346875/9e4c3dda-aa60-477b-a963-c5dfb6016bd7)

The following code is used to identify the character used as a decimal separator in the printf function. Once the float or double value is printed, this decimal separator is replaced by a period ('.').
```
static char
json_stream_gen_get_decimal_point(void)
{
    struct lconv* const p_lc = localeconv();
    if (NULL == p_lc)
    {
        return '.';
    }
    return *p_lc->decimal_point;
}
```

Reply:
Thanks, then I misunderstood the function. I thought it was printing float as a string customized to user locale, such as 5.324 -> "5,32"